test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -4,281 +4,234 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
   extend Minitest::Spec::DSL
 
   describe AssetManager::AttachmentUpdater do
-    let(:updater) { AssetManager::AttachmentUpdater }
-    let(:update_service) { mock("asset-manager-update-asset-worker") }
-    let(:attachment_data) { attachment.attachment_data }
-    let(:simple_pdf) { File.open(fixture_path.join("simple.pdf")) }
-    let(:attachment) { FactoryBot.create(:file_attachment, file: simple_pdf) }
-    let(:original) { Asset.variants[:original] }
-    let(:thumbnail) { Asset.variants[:thumbnail] }
-    let(:original_asset) { Asset.new(asset_manager_id: "original_asset_manager_id", variant: original, filename: "simple.pdf") }
-    let(:thumbnail_asset) { Asset.new(asset_manager_id: "thumbnail_asset_manager_id", variant: thumbnail, filename: "thumbnail_simple.pdf.png") }
-
-    around do |test|
-      AssetManager.stub_const(:AssetUpdater, update_service) do
-        test.call
-      end
-    end
-
-    before do
-      attachment_data.assets = [original_asset, thumbnail_asset]
-    end
-
-    context "when the attachment has been deleted" do
-      before do
-        attachment.delete
-      end
+    context "when the attachment does not belong to an attachable" do
+      it "does not update the status of any assets" do
+        attachment = create(:file_attachment)
 
-      it "does not update the asset" do
         AssetManager::AssetUpdater.expects(:call).never
 
-        updater.call(attachment_data, redirect_url: true, draft_status: true)
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, link_header: true)
       end
     end
 
-    describe "access limited updates" do
-      context "when attachment's attachable is access limited" do
-        before do
-          access_limited_object = stub("access-limited-object")
-          @organisation = create(:organisation)
-          AssetManagerAccessLimitation.stubs(:for).with(access_limited_object).returns([@organisation.content_id])
+    context "when the attachment's attachable is a draft" do
+      let(:edition) { create(:draft_news_article) }
+      let(:attachment) { create(:file_attachment, attachable: edition) }
```

Comment:
The entire context runs fine with the default factory assets, no need to generate ✅
`let(:assets) { attachment.attachment_data.assets }`